pkg/operator/staticresource/controller.go

```diff
@@ -0,0 +1,289 @@
+	c.reconciler.ReconcileResources()
+	<-stopCh
+	return nil
+}
+
+func Add(mgr manager.Manager, kubeconfig string) error {
+	// Currently the only static assets are AWS only
+	infraStatus, err := platform.GetInfraStatusUsingKubeconfig(mgr, kubeconfig)
+	if err != nil {
+		return err
+	}
+	platformType := platform.GetType(infraStatus)
+	if platformType != configv1.AWSPlatformType {
+		return nil
+	}
+
+	log.Infof("Setting up static resource controller. Platform Type is %s", platformType)
+
+	config := mgr.GetConfig()
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	controllerRef := &corev1.ObjectReference{
+		Kind:      "deployment",
+		Namespace: operatorNamespace,
+		Name:      deploymentName,
+	}
+	eventRecorder := events.NewKubeRecorder(clientset.CoreV1().Events(operatorNamespace), deploymentName, controllerRef)
+
+	r := &staticResourceReconciler{
+		clientset:     clientset,
+		logger:        log.WithFields(log.Fields{"controller": controllerName}),
+		eventRecorder: eventRecorder,
+	}
+
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	p := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isManaged(e.MetaNew)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return isManaged(e.Meta)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return isManaged(e.Meta)
+		},
+	}
+
+	cache := mgr.GetCache()
+	allFiles := append(staticFiles, templateFiles...)
+	for _, file := range allFiles {
+		objBytes := v410_00_assets.MustAsset(file)
+		obj, _, err := defaultCodec.Decode(objBytes, nil, nil)
+		if err != nil {
+			return err
+		}
+
+		informer, err := cache.GetInformer(obj)
+		if err != nil {
+			return err
+		}
+
+		err = c.Watch(&source.Informer{Informer: informer}, &handler.EnqueueRequestForObject{}, p)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isManaged(meta metav1.Object) bool {
+	// all managed resources are named pod-identity-webhook
+	if meta.GetName() == "pod-identity-webhook" && meta.GetNamespace() == operatorNamespace {
+		return true
+	}
+	// we also care about service-ca rotation
+	if meta.GetName() == "service-ca" && meta.GetNamespace() == "openshift-config-managed" {
+		return true
+	}
+	return false
+}
+
+type staticResourceReconciler struct {
+	clientset                              *kubernetes.Clientset
+	logger                                 log.FieldLogger
+	eventRecorder                          events.Recorder
+	deploymentGeneration                   int64
+	mutatingWebhookConfigurationGeneration int64
+	forceRollout                           bool
+}
+
+var _ reconcile.Reconciler = &staticResourceReconciler{}
+
+func (r *staticResourceReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.WithField("controller", controllerName).Debugf("reconciling after watch event %#v", request)
```

Comment:
Can use r.logger which already carries the "controller" field context. Several uses of this below.

Reply:
wonderful!